pkg/gardenlet/bootstrap/certificate/certificate_util.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+The utility functions in this file were copied from the kubernetes/client-go project
+https://github.com/kubernetes/client-go/blob/master/util/certificate/certificate_manager.go
+
+Modifications Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved.
+*/
+
+package certificate
+
+import (
+	"crypto/tls"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// nextRotationDeadline returns a value for the threshold at which the
+// current certificate should be rotated, 80%+/-10% of the expiration of the
+// certificate.
+func nextRotationDeadline(certificate tls.Certificate) time.Time {
+	notAfter := certificate.Leaf.NotAfter
+	totalDuration := float64(notAfter.Sub(certificate.Leaf.NotBefore))
+	deadline := certificate.Leaf.NotBefore.Add(jitteryDuration(totalDuration))
+	return deadline
+}
+
+// jitteryDuration uses some jitter to set the rotation threshold so each gardenlet
+// will rotate at approximately 70-90% of the total lifetime of the
+// certificate.  With jitter, if a number of gardenlets are added to a garden cluster at
+// approximately the same time, they won't all
+// try to rotate certificates at the same time for the rest of the lifetime
+var jitteryDuration = func(totalDuration float64) time.Duration {
```

Comment:
Why is this a `var`? Also, can you rather only convert to `float` for multiplying instead of converting back and forth for the first part?
```suggestion
func jitteryDuration(totalDuration time.Duration) time.Duration {
	return wait.Jitter(totalDuration, 0.2) - time.Duration(float64(totalDuration)*0.3)
}
```

Reply:
This is copied from [Kubernetes](https://github.com/kubernetes/client-go/blob/master/util/certificate/certificate_manager.go) as mentioned in the heading of the file. 
I do not want to change the implementation here if not absolutely necessary. 
Also I did not copy the unit tests for these utility functions as I regard them as well tested.
Though I am open to other opinions of course

Reply:
I see, I missed that.
Well, I think copying the tests makes sense in general, because we should test all code contained in our repository, but as this is only a small snippet of code, it's not so important.